client/homebrew/editor/metadataEditor/stringArrayEditor.jsx

```diff
@@ -0,0 +1,124 @@
+const React = require('react');
+const createClass = require('create-react-class');
+const _     = require('lodash');
+
+const StringArrayEditor = createClass({
+	displayName     : 'StringArrayEditor',
+	getDefaultProps : function() {
+		return {
+			label        : '',
+			values       : [],
+			valuePattern : null,
+			canEdit      : null,
+			onChange     : ()=>{}
+		};
+	},
+
+	getInitialState : function() {
+		return {
+			valueContext : !!this.props.values ? this.props.values.map((value)=>({
+				value,
+				editing : false
+			})) : [],
+			temporaryValue : '',
+			updateValue    : ''
+		};
+	},
+
+	componentDidUpdate : function(prevProps) {
+		if(!_.eq(this.props, prevProps)) {
+			this.setState({
+				valueContext : !!this.props.values ? this.props.values.map((newValue)=>({
+					value   : newValue,
+					editing : this.state.valueContext.find(({ value })=>value === newValue)?.editing || false
+				})) : []
+			});
+		}
+	},
+
+	handleChange : function(value) {
+		this.props.onChange({
+			target : {
+				value
+			}
+		});
+	},
+
+	addValue : function(value){
+		this.handleChange(_.uniq([...this.props.values, value]));
+	},
+
+	removeValue : function(value){
+		this.handleChange(this.props.values.filter((v)=>v !== value));
+	},
+
+	updateValue : function(value, index){
+		const values = this.props.values;
+		values[index] = value;
+		this.handleChange(values);
+	},
+
+	editValue : function(index){
+		if(!!this.props.canEdit && !this.props.canEdit(this.props.values[index])) {
+			return;
+		}
+		const valueContext = this.state.valueContext.map((context, i)=>{
+			context.editing = index === i;
+			return context;
+		});
+		this.setState({ valueContext, updateValue: this.props.values[index] });
+	},
+
+	handleValueInputKeyDown : function(event, index) {
+		if(event.key === 'Enter') {
```

Comment:
Pressing Enter after editing an existing value should exit edit mode for that entry. 

Also, it seems to just create a new entry but the old entry reappears when you click out.

Reply:
Clicking out as "cancel" is fine. The problem is editing and then pressing enter creates a *new* entry rather than updating the current entry, so when you click out the old entry you edited magically comes back.

Reply:
Hmm I haven't seen this behavior -- if you could send me some screenshots or a short video that would be great

Reply:
https://user-images.githubusercontent.com/5878534/152845031-fbd5f51d-c0ad-4768-908c-a2493f98f207.mp4

It might only occur when editing the first author in the list. The others seem to work fine.

Reply:
Oh that's super weird, yeah I'll figure out what's up there!

Reply:
Lmao it's because I was using the truthy operator on `index`, so when index is 0 it resolved to false and added an element instead of editing it.